hazelcast/src/main/java/com/hazelcast/config/replacer/EncryptionReplacer.java

```diff
@@ -0,0 +1,121 @@
+     * of the encryption password.
+     */
+    public static final String PROPERTY_PASSWORD_USER_PROPERTIES = "passwordUserProperties";
+    /**
+     * Replacer property name to configure network interface name used to retrieve MAC address used as part of the encryption
+     * password.
+     */
+    public static final String PROPERTY_PASSWORD_NETWORK_INTERFACE = "passwordNetworkInterface";
+    /**
+     * Replacer property name to configure path to a password file which content should be used as part of the encryption
+     * password.
+     */
+    public static final String PROPERTY_PASSWORD_FILE = "passwordFile";
+
+    private static final String PREFIX = "ENC";
+    private static final int DEFAULT_ITERATIONS = 531;
+
+    private boolean passwordUserProperties;
+    private String passwordNetworkInterface;
+    private String passwordFile;
+
+    @Override
+    public void init(Properties properties) {
+        super.init(properties);
+        passwordFile = properties.getProperty(PROPERTY_PASSWORD_FILE);
+        passwordUserProperties = Boolean.parseBoolean(properties.getProperty(PROPERTY_PASSWORD_USER_PROPERTIES, "true"));
+        passwordNetworkInterface = properties.getProperty(PROPERTY_PASSWORD_NETWORK_INTERFACE);
+        checkFalse(passwordFile == null && passwordNetworkInterface == null && !passwordUserProperties,
+                "At least one of the properties used to generate encryption password has to be configured");
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    protected char[] getPassword() {
+        try {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            if (passwordFile != null) {
+                FileInputStream fis = new FileInputStream(passwordFile);
+                try {
+                    baos.write(IOUtil.toByteArray(fis));
+                } finally {
+                    IOUtil.closeResource(fis);
+                }
+            }
+            if (passwordUserProperties) {
+                baos.write(System.getProperty("user.home").getBytes(UTF8_CHARSET));
+                baos.write(System.getProperty("user.name").getBytes(UTF8_CHARSET));
+            }
+            if (passwordNetworkInterface != null) {
+                try {
+                    NetworkInterface iface = NetworkInterface.getByName(passwordNetworkInterface);
+                    baos.write(iface.getHardwareAddress());
+                } catch (SocketException e) {
+                    throw rethrow(e);
+                }
+            }
+            return new String(Base64.encode(baos.toByteArray()), UTF8_CHARSET).toCharArray();
+        } catch (Exception e) {
+            throw rethrow(e);
+        }
+    }
+
+    public static final void main(String... args) throws Exception {
+        if (args == null || args.length < 1 || args.length > 2) {
```

Comment:
could it accept a path to hazelcast.xml - so it could read the config from the XML itself? to make sure it will use the same parameters for encryption as will be used to decrypt it?

Reply:
Nice idea, I'll work on it.

Reply:
It's implemented.